raiden/channel.py

```diff
@@ -1048,19 +1022,18 @@ def create_lockedtransfer(self, amount, identifier, expiration, hashlock):
 
         This message needs to be signed and registered with the channel before sent.
         """
+        timeout = expiration - self.block_number
+
         if not self.isopen:
             raise ValueError('The channel is closed.')
 
-        block_number = self.external_state.get_block_number()
-        timeout = expiration - block_number
-
         # the lock timeout cannot be larger than the settle timeout (otherwise
         # the smart contract cannot check the locks)
-        if timeout >= self.settle_timeout:
+        if timeout > self.settle_timeout:
```

Comment:
added a new smart contract test for that case named `test_unlock_at_settlement_block`